mutation_writer/partition_based_splitting_writer.cc

```diff
@@ -41,16 +40,15 @@ class partition_based_splitting_mutation_writer {
     bucket* _current_bucket = nullptr;
 
     future<> write_to_bucket(mutation_fragment&& mf) {
-        _current_bucket->data_size += mf.memory_usage();
         return _current_bucket->writer.consume(std::move(mf));
     }
 
     future<bucket*> create_bucket_for(const dht::decorated_key& key) {
         if (_buckets.size() < _max_buckets) {
             co_return &_buckets.emplace_back(bucket{bucket_writer(_schema, _permit, _consumer), key});
         }
-        auto it = std::max_element(_buckets.begin(), _buckets.end(), [] (const bucket& a, const bucket& b) {
-            return a.data_size < b.data_size;
+        auto it = std::max_element(_buckets.begin(), _buckets.end(), [this] (const bucket& a, const bucket& b) {
+            return dht::ring_position_tri_compare(*_schema, a.last_key, b.last_key) < 0;
```

Comment:
Let's say we have sstables with max keys 100, 200, 300, 400. If you close 400 (say because key 7 came up and you need a new bucket), and key 401 comes along, you'll have to put 401 in 300. Then if 301 comes along you'll need to put it into 200, eventually bumping 200 and 100 too.

I think bucket sorting is just doomed, let's concentrate on memtable sorting instead.